src/x_file.c

```diff
@@ -87,6 +93,30 @@ static int sys_remove(const char *pathname) {
     u8_utf8toucs2(ucs2buf, MAXPDSTRING, pathname, MAXPDSTRING);
     return do_delete_ucs2(ucs2buf);
 }
+static char* sys_getcwd(char *buf, size_t size) {
+    int err=0;
+    size_t ucs2size = size;
+    uint16_t*res = 0, *ucs2buf = getbytes(ucs2size*sizeof(*ucs2buf));
```

Comment:
why not simply use `uint16_t src[MAX_PATH]`, like in the functions above?

Reply:
good point.
the reason was file's `sys_*` closely wrap the POSIX functions (which has a `size`) parameter, and it appeared to me that this dynamic size ought to be passed thru as far as possible.

anyhow, i've now switched to a stack-based allocation.